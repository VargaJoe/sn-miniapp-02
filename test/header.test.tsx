import { mount, shallow } from 'enzyme'
import React from 'react'
import { IconButton } from '@material-ui/core'
import { act } from 'react-dom/test-utils'
import { RepositoryContext } from '@sensenet/hooks-react'
import HeaderPanel from '../src/components/header'

describe('Header', () => {
  it('Matches snapshot', () => {
    const l = shallow(<HeaderPanel />)
    expect(l).toMatchSnapshot()
  })
  it('Logout function', async () => {
    const logoutfn = jest.fn()
    const getValue = () => {
      return { DisplayName: 'Test man' } as any
    }
    const subscribe = () => {
      return { DisplayName: 'Test man', Id: 2, Domain: 'Test man' } as any
    }
    const repo = { authentication: { logout: logoutfn, currentUser: { getValue, subscribe } } }
    let wrapper: any

    await act(async () => {
      wrapper = mount(
        <RepositoryContext.Provider value={repo as any}>
          <HeaderPanel />
        </RepositoryContext.Provider>,
      )
    })

    wrapper
      .update()
      .find(IconButton)
      .simulate('click')
    expect(logoutfn).toBeCalled()
  })
})
