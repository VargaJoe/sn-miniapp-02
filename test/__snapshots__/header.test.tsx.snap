// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Matches snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-4"
        variant="h6"
      >
        Visitor
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="logout"
        className="makeStyles-logoutButton-3"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <PowerSettingsNewIcon />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
